test/e2e/storage/persistent_volumes-local.go

```diff
@@ -333,47 +312,132 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
 			Expect(err).NotTo(HaveOccurred())
 			err = framework.WaitForPersistentVolumeClaimPhase(
 				v1.ClaimBound, config.client, claim.Namespace, claim.Name, framework.Poll, 1*time.Minute)
-			Expect(claim.Spec.VolumeName).To(Equal(oldPV.Name))
 			Expect(err).NotTo(HaveOccurred())
 
+			claim, err = config.client.Core().PersistentVolumeClaims(config.ns).Get(claim.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(claim.Spec.VolumeName).To(Equal(oldPV.Name))
+
 			// Delete the persistent volume claim: file will be cleaned up and volume be re-created.
 			By("Deleting the persistent volume claim to clean up persistent volume and re-create one")
-			writeCmd, readCmd := createWriteAndReadCmds(volumePath, testFile, testFileContent)
-			_, err = framework.NodeExec(node0.Name, writeCmd)
+			writeCmd, _ := createWriteAndReadCmds(volumePath, testFile, testFileContent)
+			err = framework.IssueSSHCommand(writeCmd, framework.TestContext.Provider, config.node0)
 			Expect(err).NotTo(HaveOccurred())
 			err = config.client.Core().PersistentVolumeClaims(claim.Namespace).Delete(claim.Name, &metav1.DeleteOptions{})
 			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for a new PersistentVolume to be re-created")
 			newPV, err := waitForLocalPersistentVolume(config.client, volumePath)
 			Expect(err).NotTo(HaveOccurred())
 			Expect(newPV.UID).NotTo(Equal(oldPV.UID))
-			result, err := framework.NodeExec(node0.Name, readCmd)
+			fileDoesntExistCmd := createFileDoesntExistCmd(volumePath, testFile)
+			err = framework.IssueSSHCommand(fileDoesntExistCmd, framework.TestContext.Provider, config.node0)
 			Expect(err).NotTo(HaveOccurred())
-			Expect(result.Code).NotTo(BeZero(), "file should be deleted across local pv recreation, but exists")
 		})
 	})
 })
 
+// The tests below are run against multiple mount point types
+
+// Test two pods at the same time, write from pod1, and read from pod2
+func twoPodsReadWriteTest(config *localTestConfig, testVol *localTestVolume) {
+	By("Creating pod1 to write to the PV")
+	pod1, pod1Err := createLocalPod(config, testVol)
```

Comment:
I'm adding a sanity check to read the default test file that is placed in the volume when it is created, aka testReadDefaultFile.